gtsfm/common/view_graph.py

```diff
@@ -0,0 +1,34 @@
+"""Defines the ViewGraph data structure.
+
+Authors: Akshay Krishnan
+"""
+from dataclasses import dataclass
+from typing import Dict, Tuple, List
+
+import numpy as np
+from gtsam import Rot3, Unit3, Cal3Bundler
+
+
+@dataclass
+class ViewGraph:
+    """A data structure that stores all the two-view relationships between cameras in the scene.
+
+    Args:
+        i2Ri1: dictionary of relative rotations between camera pairs.
+        i2Ui1: dictionary of relative unit-translations between camera pairs.
+        calibrations: list of calibrations for each camera.
+        corr_idxs_i1i2: dictionary of correspondence indices between camera pairs.
+    """
+
+    i2Ri1: Dict[Tuple[int, int], Rot3]
+    i2Ui1: Dict[Tuple[int, int], Unit3]
+    calibrations: List[Cal3Bundler]
```

Comment:
Unless these are intended to be optional, I think it would be cleaner to represent this data as a `PinholeCameraCal3Bundler`.

Reply:
Why are the calibrations stored as a list while all other information is a Dict? Is it assumed that the camera IDs are contiguous and start at 0?

Reply:
Why not have absolute pose information? I think it would provide a more complete data structure.

Reply:
View-graph is meant to represent a graph forming out of 2-view relative measurements. We have one for absolute data right now: `GtsfmData`. In the future we can evaluate if we need to represent the relative and absolute measurements, but we do not have any documented use-case/to-do for this right now.

Reply:
I also am not seeing the use case currently. If it's compelling, I'm happy to rethink it